loader/src/main.rs

```diff
@@ -110,8 +125,21 @@ fn boot_sequence(args: KernelArguments, _signature: u32, fs_prehash: [u8; 64]) -
     platform::platform_tests();
 
     let mut cfg = BootConfig { base_addr: args.base as *const usize, args, ..Default::default() };
+    #[cfg(feature = "swap")]
+    println!("Size of BootConfig: {:x}", core::mem::size_of::<BootConfig>());
     read_initial_config(&mut cfg);
 
+    #[cfg(feature = "swap")]
+    {
+        cfg.swap_hal = SwapHal::new(&cfg);
+        read_swap_config(&mut cfg);
+        #[cfg(feature = "resume")]
+        println!(
+            "WARNING WARNING WARNING: swap and resume selected - this is not a valid configuration because\
```

Comment:
If it's broken, would it be a good idea to use `compile_error!()` rather than `println!()` here to make it a hard error?

Reply:
I didn't know `compiler_error!` was a thing. I'll do that.